src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,309 @@
+            {
+                appsAndFeaturesEntry.DisplayName = displayName;
+            }
+            if (appsAndFeaturesEntry.DisplayVersion.empty())
+            {
+                appsAndFeaturesEntry.DisplayVersion = displayVersion;
+            }
+            if (appsAndFeaturesEntry.Publisher.empty())
+            {
+                appsAndFeaturesEntry.Publisher = publisher;
+            }
+            if (appsAndFeaturesEntry.ProductCode.empty())
+            {
+                appsAndFeaturesEntry.ProductCode = packageId;
+            }
+
+            return appsAndFeaturesEntry;
+        }
+
+        void AppendExeExtension(std::string& value)
+        {
+            if (!HasSuffix(value, ".exe"))
+            {
+                value += ".exe";
+            }
+        }
+
+        bool AddPortableLinksDirToPathRegistry(Manifest::ScopeEnum& scope, Utility::Architecture& arch, bool& isPathModified)
+        {
+            Key key;
+            if (scope == Manifest::ScopeEnum::Machine)
+            {
+                key = Registry::Key::CreateKeyAndOpen(HKEY_LOCAL_MACHINE, Normalize(s_PathSubkey_Machine));
+            }
+            else
+            {
+                key = Registry::Key::CreateKeyAndOpen(HKEY_CURRENT_USER, Normalize(s_PathSubkey_User));
+            }
+
+            std::string portableLinksDir = GetPortableLinksLocation(scope, arch).u8string();
+            AICLI_LOG(CLI, Info, << "Adding to Path environment variable: " << portableLinksDir);
+            std::wstring pathKey = { L"Path" };
+            std::string pathValue = key[pathKey]->GetValue<Value::Type::String>();
+            if (pathValue.find(portableLinksDir) == std::string::npos)
+            {
+                std::string modifiedPathValue = pathValue;
+
+                if (modifiedPathValue.back() != ';')
+                {
+                    modifiedPathValue += ";";
+                }
+
+                modifiedPathValue += portableLinksDir + ";";
+                bool result = key.SetKeyValue(pathKey, ConvertToUTF16(modifiedPathValue), REG_EXPAND_SZ);
+                if (result)
+                {
+                    isPathModified = true;
+                }
+            }
+
+            return true;
+        }
+
+        bool AddPortableEntryToUninstallRegistry(Manifest::ScopeEnum& scope, AppsAndFeaturesEntry& entry)
+        {
+            HKEY root = (scope == Manifest::ScopeEnum::Machine) ? HKEY_LOCAL_MACHINE : HKEY_CURRENT_USER;
+            std::wstring productCode = ConvertToUTF16(entry.ProductCode);
+            std::wstring displayName = ConvertToUTF16(entry.DisplayName);
+            std::wstring displayVersion = ConvertToUTF16(entry.DisplayVersion);
+            std::wstring publisher = ConvertToUTF16(entry.Publisher);
+            std::wstring uninstallString = L"winget uninstall --id " + productCode;
+            std::wstring fullRegistryKey = Normalize(s_UninstallSubkey) + L"\\" + productCode;

```

Comment:
This doesn't appear to be taking the architecture into account when appropriate.

Reply:
Along these same lines, would it be best to abstract the registry functions to their own class? I'm thinking of other use cases outside of portable apps like package pinning or for tracking overrides, one way that it could be done is by adding keys to the app's uninstall  registry. Rather than having each workflow have its own unique registry paths, having a single namespace for winget registry updates might be beneficial in the future